### PR TITLE
Fix annotations if value is null

### PR DESCRIPTION
**Pull request by aljesusg**
Fix https://github.com/kiali/kiali/issues/5864

### CODE REVIEW

**Comment by aljesusg** [comment]
> Now application works but server still returns null value on annotations field. Should not return empty [] when there are not annotations on the service?

should... but let's merge this one to fix the error in the front, then we can discuss about this because I expect more "return null values" in other sides

**Comment by ferhoyos** [comment]
ok, in that case frontend code lgtm. Approved.

**Review by ferhoyos — APPROVED** [review]

**Comment by ferhoyos** [comment]
@aljesusg Upss I do not have write access yet. You will need a maintainer approval to be able to merge the PR.

**Comment by aljesusg** [comment]
> @aljesusg Upss I do not have write access yet. You will need a maintainer approval to be able to merge the PR.

step by step but at least you are reviewing ;)

**Comment by hhovsepy** [comment]
It seems when user edits and removes all annotations, then it is not possible to add them anymore as the "Edit Annotations" action item is removed. Which causes inconsistency:
Screenshot from 2023-02-21 18-46-01

@aljesusg can you include also backend changes into this PR to have consistency?

File: frontend/src/components/IstioWizards/ServiceWizardDropdown.tsx
Comment (hhovsepy) [inline frontend/src/components/IstioWizards/ServiceWizardDropdown.tsx:259]: When there is not annotation, then it causes dead end.

**Comment by aljesusg** [comment]
> It seems when user edits and removes all annotations, then it is not possible to add them anymore as the "Edit Annotations" action item is removed. Which causes inconsistency: Screenshot from 2023-02-21 18-46-01
> 
> @aljesusg can you include also backend changes into this PR to have consistency?

Yes...it makes sense...that is why it should return the backend and empty array let me make the change

**Comment by hhovsepy** [comment]
@aljesusg when backend is fixed, can the frontend null checks be removed on annotations for Service and Workloads? So the Edit Annotations are back.

**Comment by hhovsepy** [comment]
Here I will suggest this change on line https://github.com/kiali/kiali/blob/master/frontend/src/components/IstioWizards/WizardAnnotations.tsx#L47 because the original issue with page crashing is back.
`Object.keys(this.props.annotations || {}).map((value, index) => m.set(index, [value, this.props.annotations[value]]));`

**Review by hhovsepy — APPROVED** [review]
Verified now. The failing cypress tests are flakes.

**Review by ferhoyos — APPROVED** [review]
Verified from my end too.

**Pull request merged**
